system-test/logging.js

```diff
@@ -0,0 +1,457 @@
+      logging.getServices({ pageSize: 1 })
+        .on('error', done)
+        .once('data', function() {
+          this.end();
+          done();
+        });
+    });
+
+    it('should get indexes', function(done) {
+      service.getIndexes({ pageSize: 1 }, function(err, indexes) {
+        assert.ifError(err);
+        assert.strictEqual(indexes.length, 1);
+        done();
+      });
+    });
+
+    it('should list indexes as a stream', function(done) {
+      service.getIndexes({ pageSize: 1 })
+        .on('error', done)
+        .on('data', util.noop)
+        .on('end', done);
+    });
+  });
+
+  describe('sinks (user auth operations)', function() {
+    beforeEach(function() {
+      if (!loggingWithUserAuth) {
+        this.skip();
+      }
+    });
+
+    it('should create a sink with a Bucket destination', function(done) {
+      var sink = loggingWithUserAuth.sink(generateName());
+
+      sink.create({
+        destination: bucket
+      }, function(err, sink, apiResponse) {
+        assert.ifError(err);
+
+        var destination = 'storage.googleapis.com/' + bucket.name;
+        assert.strictEqual(apiResponse.destination, destination);
+
+        sink.delete(done);
+      });
+    });
+
+    it('should create a sink with a Dataset destination', function(done) {
+      var sink = loggingWithUserAuth.sink(generateName());
+
+      sink.create({
+        destination: dataset
+      }, function(err, sink, apiResponse) {
+        assert.ifError(err);
+
+        var destination = format('{baseUrl}/projects/{pId}/datasets/{dId}', {
+          baseUrl: 'bigquery.googleapis.com',
+          pId: dataset.parent.projectId,
+          dId: dataset.id
+        });
+
+        assert.strictEqual(apiResponse.destination, destination);
+
+        sink.delete(done);
+      });
+    });
+
+    it('should create a sink with a Topic destination', function(done) {
+      var sink = loggingWithUserAuth.sink(generateName());
+
+      sink.create({
+        destination: topic
+      }, function(err, sink, apiResponse) {
+        assert.ifError(err);
+
+        var destination = 'pubsub.googleapis.com/' + topic.name;
+        assert.strictEqual(apiResponse.destination, destination);
+
+        sink.delete(done);
+      });
+    });
+
+    describe('metadata', function() {
+      var sink;
+
+      before(function(done) {
+        if (!loggingWithUserAuth) {
+          this.skip();
+          return;
+        }
+
+        sink = loggingWithUserAuth.sink(generateName());
+
+        sink.create({
+          destination: topic
+        }, done);
+      });
+
+      beforeEach(function() {
+        if (!loggingWithUserAuth) {
+          this.skip();
+        }
+      });
+
+      after(function(done) {
+        if (!loggingWithUserAuth) {
+          this.skip();
+          return;
+        }
+
+        sink.delete(done);
+      });
+
+      it('should set metadata', function(done) {
+        var metadata = {
+          filter: 'name:*'
```

Comment:
those are unit tests, or against a live system?
if the later, this should fail, as that is an invalid filter
(filter must match an existing LogEntry field).